### PR TITLE
fix(workspaces): mutate on join workspace

### PR DESCRIPTION
**Pull request by cdriesler**
## Description & motivation

- Use new utils to also mutate on join

## Checklist:

- [ ] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [ ] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [ ] My commits are related to the pull request and do not amend unrelated code or documentation.
- [ ] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

File: packages/frontend-2/components/workspace/invite/DiscoverableWorkspaceBanner.vue [hunk 1]
```diff
@@ -60,13 +65,31 @@ const processJoin = async (accept: boolean) => {
...
   const result = await apollo
     .mutate({
       mutation: DashboardJoinWorkspaceDocument,
       variables: {
         input: {
           workspaceId: props.workspace.id
         }
+      },
+      update(cache) {
+        modifyObjectField(
```
Comment (fabis94) [inline packages/frontend-2/components/workspace/invite/DiscoverableWorkspaceBanner.vue:80]: You should always use the 2nd param of `update` - `data` - to get the actual data from the response and validate that the mutation did in fact go through successfully before you make any cache updates

**Review by fabis94 — CHANGES_REQUESTED** [review]

File: packages/frontend-2/components/workspace/invite/DiscoverableWorkspaceBanner.vue [hunk 2]
```diff
@@ -60,13 +65,31 @@ const processJoin = async (accept: boolean) => {
...
+          cache,
+          getCacheId('User', userId),
+          'workspaces',
+          ({ helpers: { createUpdatedValue, ref } }) => {
+            return createUpdatedValue(({ update }) => {
```
Comment (fabis94) [inline packages/frontend-2/components/workspace/invite/DiscoverableWorkspaceBanner.vue:85]: @cdriesler User.workspaces has variables as well, you usually want to just evict all filtered (e.g. search specified) cache values cause there's no way to accurately tell if the new workspace fits that filter or not, and then make the cache update to the list only in non-filtered values

Comment (fabis94) [reply packages/frontend-2/components/workspace/invite/DiscoverableWorkspaceBanner.vue:85]: Evict means - wherever that evicted value was referenced, e.g. in some kind of query living in a UI component, the query will understand that its now missing a part of its data and refetch it

**Comment by cdriesler** [comment]
Ok, updated, but now it's pretty similar to the mutation when creating a workspace. Acceptable for now?

**Review by fabis94 — CHANGES_REQUESTED** [review]
i think u need to merge main, there's conflicts and it shows my changes here

**Review by fabis94 — APPROVED** [review]

**Pull request merged**